src/main/java/com/paskar/email/application/console/InformationFromConsole.java

```diff
@@ -0,0 +1,113 @@
+package com.paskar.email.application.console;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.paskar.email.application.repositiory.CreateAndSaveEmail;
+import org.springframework.stereotype.Component;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Component
+public class InformationFromConsole implements CreateAndSaveEmail {
```

Comment:
This component is mixing two different types of logic inside: working with storage and working with console. This makes it less stable than it could be because it has two different reasons to change: you change the way you gather data from user(switch from console to web, for instance) and changing storage mechanism(migrate from files to database). So if you want to support console and web api you will need to create two implementations of the interface, but implementation of the storage does not need to change. Now, if you want to support file storage and database storage you again will need to create two implementations where working with console(or web) will be the same. If we sum it up, this approach leads you to code duplication and it's harder to maintain. So my suggestion is to create an interface which is responsible only for storage(and by storage I mean saving and retrieving information). Working with console should be extracted into different component